File: repos_estudiantes_ALVAREZ_primera_entrega/CALLE_CASTILLO_CRISTHIAN_JAVIER/PRACTICA1-PROGRAMACION-II/PRACTICA INF-121/PROGRAMACION ORIENTADA A OBJETOS/EJERCICIO 1/Persona.java
public class Persona {
    String nombre;
    int edad;
    String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    public void saludar() {
        System.out.println("Hola, soy " + nombre + " de " + ciudad);
    }

    public boolean esMayorDeEdad() {
        return edad >= 18;
    }

    public static void main(String[] args) {
        Persona persona1 = new Persona("Ana", 22, "La Paz");
        Persona persona2 = new Persona("Luis", 17, "Cochabamba");
        Persona persona3 = new Persona("Maria", 30, "Santa Cruz");

        Persona[] personas = {persona1, persona2, persona3};

        for (Persona p : personas) {
            p.saludar();
            if (p.esMayorDeEdad()) {
                System.out.println(p.nombre + " es mayor de edad.");
            } else {
                System.out.println(p.nombre + " NO es mayor de edad.");
            }
        }
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/CALLE_CASTILLO_CRISTHIAN_JAVIER/PRACTICA1-PROGRAMACION-II/PRACTICA INF-121/PROGRAMACION ORIENTADA A OBJETOS/EJERCICIO 1/Persona.py
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad

    def saludar(self):
        print(f"Hola, soy {self.nombre} de {self.ciudad}")

    def es_mayor_de_edad(self):
        return self.edad >= 18

persona1 = Persona("Ana", 22, "La Paz")
persona2 = Persona("Luis", 17, "Cochabamba")
persona3 = Persona("Maria", 30, "Santa Cruz")

personas = [persona1, persona2, persona3]
for persona in personas:
    persona.saludar()
    if persona.es_mayor_de_edad():
        print(f"{persona.nombre} es mayor de edad.")
    else:
        print(f"{persona.nombre} NO es mayor de edad.")
